src/medCore/database/medDatabaseExporter.cpp

```diff
@@ -59,7 +73,24 @@ void medDatabaseExporter::internalRun(void)
     }
 
     dtkAbstractDataWriter * dataWriter = medAbstractDataFactory::instance()->writer(d->writer);
-    dataWriter->setData(d->data);
+    if(!d->saveMultipleData)
+    {
+        dataWriter->setData(d->data);
+    }
+    else
+    {
+        medAbstractDataWriter* medDataWriter;
+        try
+        {
+            medDataWriter = dynamic_cast<medAbstractDataWriter*>(dataWriter);
```

Comment:
You won't get any exception raised here, so no need for a try-catch. What you want to do is check that medDataWriter != NULL.